btree/btree.c

```diff
@@ -5013,6 +5412,56 @@ static void vkvv_move_dir(struct slot *slot)
 	}
 }
 
+static void vkvv_indirect_key_fill(struct slot *slot)
+{
+	void     **p_key_addr = vkvv_key(slot->s_node, slot->s_idx);
+	void      *key_addr   = slot->s_rec.r_key.k_data.ov_buf[0];
+	uint32_t  *key_ref;
+
+	*p_key_addr = key_addr;
+	key_ref     = key_addr + INDIRECT_REF_COUNT_OFFSET;
+	*key_ref    = *key_ref + 1;
+}
+
+static void vkvv_indirect_val_fill(struct slot *slot)
+{
+	void **p_val_addr = vkvv_val(slot->s_node, slot->s_idx + 1);
+	void  *val_addr   = slot->s_rec.r_val.ov_buf[0];
+
+	*p_val_addr = val_addr;
+}
+
+static void vkvv_indirect_rec_make(struct slot *slot)
+{
+	struct vkvv_head *h     = vkvv_data(slot->s_node);
+	int               count = h->vkvv_used;
+	uint32_t          ksize;
+	uint32_t          vsize;
+	void             *start_key_addr;
+	void             *start_val_addr;
+	uint32_t          total_ksize;
+	uint32_t          total_vsize;
+
+	if (slot->s_idx == count)
+		goto free_indirect_key;
+
+	start_key_addr = vkvv_key(slot->s_node, slot->s_idx);
+	start_val_addr = vkvv_val(slot->s_node, count);
+	ksize = INDIRECT_KEY_SIZE;
+	vsize = INDIRECT_VAL_SIZE;
+
+	total_ksize = ksize * (h->vkvv_used - slot->s_idx);
+	total_vsize = vsize * (h->vkvv_used - slot->s_idx);
+
+	m0_memmove(start_key_addr + ksize, start_key_addr, total_ksize);
+	m0_memmove(start_val_addr - vsize, start_val_addr , total_vsize);
+
+	free_indirect_key:
```

Comment:
```suggestion
	if (slot->s_idx != count) {
	start_key_addr = vkvv_key(slot->s_node, slot->s_idx);
	start_val_addr = vkvv_val(slot->s_node, count);
	ksize = INDIRECT_KEY_SIZE;
	vsize = INDIRECT_VAL_SIZE;

	total_ksize = ksize * (h->vkvv_used - slot->s_idx);
	total_vsize = vsize * (h->vkvv_used - slot->s_idx);

	m0_memmove(start_key_addr + ksize, start_key_addr, total_ksize);
	m0_memmove(start_val_addr - vsize, start_val_addr , total_vsize);
}
```

Reply:
Is it because to avoid go-to statements as it breaks the flow?